### PR TITLE
Fix read-only mount flag after restart

### PR DESCRIPTION
**Pull request by madanadit**

### CODE REVIEW

**Comment by madanadit** [comment]
I'll add a test case as well

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 1]
```diff
@@ -561,7 +561,9 @@ public void start(Boolean isPrimary) throws IOException {
         MountInfo mountInfo = mMountTable.getMountTable().get(key);
         UnderFileSystemConfiguration ufsConf =
             UnderFileSystemConfiguration.defaults(ServerConfiguration.global())
-            .createMountSpecificConf(mountInfo.getOptions().getPropertiesMap());
+                .createMountSpecificConf(mountInfo.getOptions().getPropertiesMap())
```
Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: The API here is pretty confusing, it is not obvious we need to set the read only or shared fields separately.
For the fix to released branches I'm fine with this, to master we should should consider something else. How about a `fromMountInfo` method?

Comment (madanadit) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: makes sense. not targeting this PR though as its a big refactor

**Comment by madanadit** [comment]
jenkins, test this please.

**Comment by shaneknapp** [comment]
failure caused by testing out a new build node

**Comment by shaneknapp** [comment]
test this please

**Comment by madanadit** [comment]
jenkins, test this please.

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java (hunk 1 above)
Comment (ZacBlanco) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: Can we file a tech-debt issue for this?

Comment (madanadit) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: done

**Comment by madanadit** [comment]
@ZacBlanco do you have more comments?

**Comment by ZacBlanco** [comment]
Can you link the issue to this thread/PR? I can't seem to find it.

Other than that, LGTM

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Comment by madanadit** [comment]
alluxio-bot, cherry-pick this to master please.

**Pull request merged**